File: selenium/src/java/org/openqa/selenium/internal/seleniumemulation/ShiftKeyUp.java
package org.openqa.selenium.internal.seleniumemulation;

import org.openqa.selenium.WebDriver;

public class ShiftKeyUp extends SeleneseCommand<Void> {
  private final KeyState keyState;

  public ShiftKeyUp(KeyState keyState) {
    this.keyState = keyState;
  }

  @Override
  protected Void handleSeleneseCommand(WebDriver driver, String locator, String value) {
    keyState.shiftKeyDown = false;
    return null;
  }
}

File: selenium/src/csharp/WebdriverBackedSelenium/Internal/SeleniumEmulation/ShiftKeyUp.cs
using System;
using System.Collections.Generic;
using System.Text;
using OpenQA.Selenium;

namespace Selenium.Internal.SeleniumEmulation
{
    internal class ShiftKeyUp : SeleneseCommand
    {
        private KeyState keyState;

        public ShiftKeyUp(KeyState keyState)
        {
            this.keyState = keyState;
        }

        protected override object HandleSeleneseCommand(IWebDriver driver, string locator, string value)
        {
            keyState.ShiftKeyDown = true;
            return null;
        }
    }
}
